presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -236,8 +237,13 @@ public PlanNode visitGroupId(GroupIdNode node, RewriteContext<Expression> contex
         @Override
         public PlanNode visitMarkDistinct(MarkDistinctNode node, RewriteContext<Expression> context)
         {
-            checkState(!SymbolsExtractor.extractUnique(context.get()).contains(node.getMarkerSymbol()), "predicate depends on marker symbol");
-            return context.defaultRewrite(node, context.get());
+            if (!SymbolsExtractor.extractUnique(context.get()).contains(node.getMarkerSymbol())) {
+                return context.defaultRewrite(node, context.get());
+            }
+            if (context.get() == TRUE_LITERAL) {
+                return node;
+            }
+            return new FilterNode(idAllocator.getNextId(), node, context.get());
```

Comment:
please address: https://github.com/prestodb/presto/pull/8435#discussion_r179262608

Example, `markDistinctSymbols=[a]`,
rows:
```
a|b
1|3
1|2
``` 
predicate: `b=2`, plan:
```
Filter[b=2]
 |
MarkDistinct[a]
```

Result without pushing predicate down:
```
1,2,false
```
Result with pushing predicate down:
```
1,2,true
```

As you can see result is changed. This is because MarkDistinct is kind of "undeterministic" on symbols other than `distinctSymbols`.

Reply:
I see your point, but isn't this kind of query which is know to be non deterministic? Like `SELECT * FROM t LIMIT 1` is also non deterministic, it may return random row.

Anyway I will follow your suggestion for now.